frontend/test/__support__/server-mocks/search.ts

```diff
@@ -1,21 +1,37 @@
 import fetchMock from "fetch-mock";
-import type { CollectionItem } from "metabase-types/api";
-
-export function setupSearchEndpoints(items: CollectionItem[]) {
-  const availableModels = items.map(({ model }) => model);
+import type { CollectionItem, SearchResult } from "metabase-types/api";
 
+export function setupSearchEndpoints(items: (CollectionItem | SearchResult)[]) {
   fetchMock.get("path:/api/search", uri => {
     const url = new URL(uri);
-    const models = url.searchParams.getAll("models");
+    const models = [...new Set(url.searchParams.getAll("models"))];
     const limit = Number(url.searchParams.get("limit"));
     const offset = Number(url.searchParams.get("offset"));
     const table_db_id = url.searchParams.get("table_db_id") || null;
-    const matchedItems = items.filter(({ model }) => models.includes(model));
+    const queryText = url.searchParams.get("q");
+
+    let matchedItems = items;
+    if (queryText) {
+      const searchString = queryText.toLowerCase();
+
+      matchedItems = matchedItems.filter(({ name }) =>
+        name.toLowerCase().includes(searchString),
+      );
+    }
+
+    // available_models: all possible types in a user's database (or in this case, the test data), AFTER the text query filter
+    const availableModels = [
```

Comment:
I think this could be further simplified 

```
const models = url.searchParams.getAll("models");
const queryText = url.searchParams.get("q").toLowerCase();

const matchedItems =  items
  .filter(({ name }) => name.toLowerCase().includes(queryText)
  .filter(({ model } => !models.length || models.includes(model))
  
```

Reply:
`Set` seems unneeded btw

Reply:
Gotcha - I need to split the filtering here so I can grab `availableModels` based on the text search query, but I've added everything else